lib/pavilion/expression_functions/core.py

```diff
@@ -511,3 +511,44 @@ def outliers(values: List[num], names: List[str], limit: float):
                 deviations[names[i]] = dev
 
         return deviations
+
+class FlagPlugin(CoreFunctionPlugin):
+    """Convert a boolean(-ish) config option into a UNIX-style flag.
+    """
+
+    def __init__(self):
+        super().__init__(
+            'flag',
+            arg_specs=(str, str) # Should this first arg be a str or a bool?
+        )
+
+    @staticmethod
+    def flag(val: str, flag_str: str):
+        return flag(val, flag_str)
+
+class SoptPlugin(CoreFunctionPlugin):
+    """Convert a config option into a UNIX-style option string. If the
+    option is a list, create a separate option string for each element."""
+
+    def __init__(self):
+        super().__init__(
+            'sopt',
+            arg_specs=(MaybeList(str), str)
+        )
+
+    @staticmethod
+    def sopt(val: Union[str, List[str]], opt_str: str) -> str:
+        return sopt(val, opt_str)
+
+class OptPlugin(CoreFunctionPlugin):
+    """Convert a config option into a UNIX-style option string. If the option is
+    a list, concatenate the elements of the list to form a single option."""
+    def __init__(self):
+        super().__init__(
+            'opt',
+            arg_specs=(MaybeList(str), str, str)
+        )
+
+    @staticmethod
+    def opt(val: Union[str, List[str]], opt_str: str, delimiter: str = ',') -> str:
+        return opt(val, opt_str, delimiter)
```

Comment:
The same here on the order, and the last argument can be optional `arg_specs=(str, MaybeList(str), Opt(str))`